### PR TITLE
Update DG

### LINKED ISSUE #262: Update DG (user stories)
update user stories to match current functionality

### LINKED ISSUE #263: Update inaccurate use case descriptions
image

### LINKED ISSUE #264: DG: Edit formatting
image

and make Model diagram larger

### LINKED ISSUE #294: [PE-D][Tester B] Typo in DG
Typo in DG under the section for Vendor implementation, step 3

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: YANYAN-w/ped#8

### LINKED ISSUE #354: DG: Update parser/command UML

### PR DESCRIPTION
**Pull request by tingxuanp**
Closes #294 
Closes #263 
Closes #354 
Closes #264 
Closes #262

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -349,7 +349,7 @@ Step 1. The user launches the application, `Persons` and `Vendors` are loaded in
 
 Step 2. The user executes `assign-vendor 1`. WedLinker will assign the `Person` at index 1 of the current displayed person list to become a `Vendor`. The user can now assign `Tasks` to this contact.
 
-Step 3. The user executes `unassign-vendor 1` to unassign the `Vendor` to become a non-. WedLinker includes the `Person` in the `Guest List`.
+Step 3. The user executes `unassign-vendor 1`. The contact has been is no longer a `Vendor`.
```
Comment (DanzaSeah) [inline docs/DeveloperGuide.md:352]: Maybe i'm misunderstanding, but the I don't quite understand this,  is it supposed to say this?

>  The contact is no  longer a `Vendor`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -562,7 +562,7 @@ Similar to [<ins>UC01](#use-case-uc01-list-all-contacts) except to view tasks in
 Similar to [<ins>UC04](#use-case-uc04-create-a-contact) except adding a wedding to WedLinker instead of a Contact
 
 ### **Use case: UC06 — Create a Task**
-Similar to [<ins>UC04](#use-case-uc04-create-a-contact) except adding a task to WedLinker instead of a Contact
+Similar to [<ins>UC04]( #use-case-uc04-create-a-contact) except adding a task to WedLinker instead of a Contact
```
Comment (DanzaSeah) [inline docs/DeveloperGuide.md:565]: Just wondering, is there a particular reason for the added space, or maybe it was an unintentional addition

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -899,23 +892,24 @@ feature freeze. The plans to improve our features are as such.
 
 1. There is a small GUI bug that causes the overflow of names for Tags and Weddings which causes cosmetic flaws.
 Planned enhancements would be to truncate the name with `...`.
-2. There is a duplicate validation bug that allows the creation of certain "duplicate" Person, Wedding, Tag, Task, Tags.
+2. There is a duplicate validation bug that allows the creation of certain "duplicate" Person, Wedding, Task and Tags.
```
Comment (DanzaSeah) [inline docs/DeveloperGuide.md:895]: Just wondering, is there a reason for everything to be singular except for Tags?

**Review by DanzaSeah — COMMENTED** [review]
LGTM, just some nits but otherwise looks good

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (tingxuanp) [reply docs/DeveloperGuide.md]: I think it was just a typo, thanks for pointing out

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (tingxuanp) [reply docs/DeveloperGuide.md:565]: yep unintentional, thanks for pointing out!

**Review by DanzaSeah — COMMENTED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -470,12 +471,12 @@ Those without any stars are user stories that were considered but will not be im
...
-| `* * *`  | user                      | create tags                                                                                                          | I can have special categories for non traditional vendors.                                                                                                          |
-|  `* *`   | user                      | edit information such as the contact number and address of each contact                                              | all contacts have the most updated information.                                                                                                                     |
+| `* * *`  | user                      | search for contacts by name, address, phone number, email                                                            | I can find specific contacts that I am looking for.                                                                                                                 |
```
Comment (dasha3412) [inline docs/DeveloperGuide.md:477]: Can maybe add wedding here, or add as separate user story?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -484,16 +485,16 @@ Those without any stars are user stories that were considered but will not be im
...
-|   `*`    | user                      | assign tasks to contacts                                                                                             | I can track which tasks have been assigned to each contact.                                                                                                         |
-|   `*`    | user                      | update the status of tasks of contacts                                                                               | I can track the status of completion of the tasks assigned to contacts.                                                                                             |
+|  `* *`   | lazy user                 | want to use command keyword shortcuts                                                                                | I can quickly execute commands without needing to type the full command word.                                                                                       |
```
Comment (dasha3412) [inline docs/DeveloperGuide.md:488]: I think for second part of this, can maybe rephrase to: "can use command...". Also, I think the use case this replaces (lazy user | have the ability to bypass certain controlled features) is more referring to the force flag? So maybe we should keep that one

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -484,16 +485,16 @@ Those without any stars are user stories that were considered but will not be im
...
+|  `* *`   | user                      | assign tasks to certain contacts                                                                                     | I can track which tasks have been assigned to each contact.                                                                                                         |
+|  `* *`   | user                      | update the status of tasks of contacts                                                                               | I can track the status of completion of the tasks assigned to contacts.                                                                                             |
```
Comment (dasha3412) [inline docs/DeveloperGuide.md:490]: Maybe can just add 'update the **completion** status of tasks ...', and change the 'so that' section to say something like 'I can track which tasks each contact has completed or must still complete.' just to avoid repeating the 'I can' section

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -484,16 +485,16 @@ Those without any stars are user stories that were considered but will not be im
...
+|  `* *`   | busy user                 | add multiple wedding events                                                                                          | I can track contacts for multiple weddings at once.                                                                                                                 |
+|  `* *`   | busy user                 | tag each contact to a wedding                                                                                        | I can easily see which contacts are relevant to each wedding.                                                                                                       |
```
Comment (dasha3412) [inline docs/DeveloperGuide.md:492]: Maybe here for the "I can", can say "assign each..." just to avoid confusion with tags?

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -484,16 +485,16 @@ Those without any stars are user stories that were considered but will not be im
...
-|   `*`    | user                      | assign dates to a wedding                                                                                            | I can keep track of when different weddings are scheduled.                                                                                                          |
-|   `*`    | user                      | filter contacts by wedding                                                                                           | I can keep track of which contacts are relevant for each wedding.                                                                                                   |
+|   `*`    | user                      | assign a rating out of 5 to each vendor                                                                              | I can track the experience with this vendor for future reference.                                                                                                   | |
```
Comment (dasha3412) [inline docs/DeveloperGuide.md:497]: Is there a special notation we use to say that we won't be completing a user story? I think might need to document this as one of them

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -605,32 +606,25 @@ Similar to [<ins>UC07](#use-case-uc07-add-phone-number-to-contact) except duplic
...
+### **Use case: UC14 — Search for Contacts by Tag**
+Similar to [<ins>UC10](#use-case-uc10-search-for-contacts-by-name) except searching for contacts by tag.
 
-### **Use case: UC17 — Search for Task by Task Name**
-Similar to [<ins>UC10](#use-case-uc10-search-for-contacts-by-name) except searching for task by task name.
+### **Use case: UC15 — Search for Contacts by Wedding**
+Similar to [<ins>UC10](#use-case-uc10-search-for-contacts-by-name) except searching for contacts by wedding.
 
----
+### **Use case: UC16 — Search for Contacts by Task**
```
Comment (dasha3412) [inline docs/DeveloperGuide.md:624]: Can maybe add use case for searching for contacts by tag?

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -894,28 +888,40 @@ testers are expected to do more *exploratory* testing.
...
+align with the case sensitivity defined for Tag in the `Tag::isSameTag(Tag)` function, nor does it reflect real-world case-sensitivity of tag names.
+The planned enhancement would be to ensure that when untagging Person objects, case is ignored and the `Tag::isSameTag(Tag)` function is used
+to check for ensure proper matching of Tag regardless of case.
+
+6. **Allow searching contacts with blank fields**: Currently, the `find` command does not support searching for contacts with blank field values.
+For example, using the command `find a/`, where the address field is empty, will display an error on the GUI telling the user that the field to be searched cannot be empty. 
+However, contacts can currently be added with all fields, other from name, blank. Thus, there may be valid cases where users want to search for contacts with certain blank fields. 
+The planned enhancement would be to modify the `find` command to allow users to search for contacts with blank fields.
+When no keywords are provided after a prefix (e.g., `find a/`), the application should return a list of contacts where the corresponding field is blank.
+
```
Comment (dasha3412) [inline docs/DeveloperGuide.md:927]: Sorry to have not mentioned this earlier, but another planned enhancement I believe is adding processing for multiple partner tags in the assign-wedding instruction that considers the order of the p1/ and p2/ tags or rejects the command if there are multiple partner tags specified for one wedding.

Also, I think UI auto-updating after certain commands (I think unassign-wedding, for example) might be a planned enhancement? I'm actually not sure if this qualifies as planned enhancement

**Review by dasha3412 — COMMENTED** [review]
Updates look great!

You mentioned in your PR summary that this also resolves #264 although I'm not too sure what the error is there, but I can't see any updates to the edit user story - is this PR meant to resolve that?

Also, I think we have more user stories we need to add based on things like tasks, tags, vendors, weddings, etc. to resolve #262 ?

For these, happy to approve and merge first and then have these other issues be fixed in another PR, just need to remove from description.
